backend/census_historical_migration/test_notes_to_sefa_xforms.py

```diff
@@ -226,3 +235,253 @@ def test_note_with_title_content(self):
         for note in result:
             self.assertNotIn(settings.GSA_MIGRATION, note.TITLE)
             self.assertNotIn(settings.GSA_MIGRATION, note.CONTENT)
+
+
+class TestXformMissingNotesRecordsV2(SimpleTestCase):
+    class MockAuditHeader:
+        def __init__(
+            self,
+            DBKEY,
+            AUDITYEAR,
+        ):
+            self.DBKEY = DBKEY
+            self.AUDITYEAR = AUDITYEAR
+
+    class MockNote:
+        def __init__(
+            self,
+            TITLE,
+            CONTENT,
+            TYPE_ID,
+            DBKEY,
+            AUDITYEAR,
+        ):
+            self.TITLE = TITLE
+            self.CONTENT = CONTENT
+            self.TYPE_ID = TYPE_ID
+            self.DBKEY = DBKEY
+            self.AUDITYEAR = AUDITYEAR
+
+    def _mock_notes_valid_year_policies_no_rate(self):
+        notes = []
+        notes.append(
+            self.MockNote(
+                TITLE="Significant Accounting Policies Used in Preparing the SEFA",
+                CONTENT="Expenditures reported on the Schedule are reported on the accrual basis of accounting.  Such expenditures are recognized following the cost principles contained in the Uniform Guidance, wherein certain types of expenditures are not allowable or are limited as to reimbursement.",
+                TYPE_ID=1,
+                DBKEY="123456789",
+                AUDITYEAR="2022",
+            )
+        )
+        audit_header = self.MockAuditHeader(
+            DBKEY="123456789",
+            AUDITYEAR="2022",
+        )
+        return notes, audit_header
+
+    def _mock_notes_valid_year_rate_no_policies(self):
+        notes = []
+        notes.append(
+            self.MockNote(
+                TITLE="10% De Minimis Cost Rate",
+                CONTENT="The auditee did not elect to use the de minimis cost rate.",
+                TYPE_ID=2,
+                DBKEY="223456789",
+                AUDITYEAR="2021",
+            )
+        )
+        audit_header = self.MockAuditHeader(
+            DBKEY="223456789",
+            AUDITYEAR="2021",
+        )
+        return notes, audit_header
+
+    def _mock_notes_valid_year_no_rate_no_policies(self):
+        notes = []
+        audit_header = self.MockAuditHeader(
+            DBKEY="223456788",
+            AUDITYEAR="2020",
+        )
+        return notes, audit_header
+
+    def _mock_notes_valid_year_rate_policies(self):
+        notes = []
+        notes.append(
+            self.MockNote(
+                TITLE="10% De Minimis Cost Rate",
+                CONTENT="The auditee did not elect to use the de minimis cost rate.",
+                TYPE_ID=2,
+                DBKEY="123456788",
+                AUDITYEAR="2018",
+            )

```

Comment:
To avoid repeating these MockNotes, thoughts on making a policy const and a rate const?

Reply:
There are three variables for xform_missing_notes_records_v2 - audit_year, policies and rate.  Every mock record has different (DBKEY, AUDITYEAR) values to allow for variations.

Reply:
Sorry, I meant the policy/rate part. Something like this defined outside the methods:
```
const rate = {
                TITLE="10% De Minimis Cost Rate",
                CONTENT="The auditee did not elect to use the de minimis cost rate.",
                TYPE_ID=2,
                DBKEY="123456788",
                AUDITYEAR="2018",
}
const policy = {
                TITLE="Significant Accounting Policies Used in Preparing the SEFA",
                CONTENT="Expenditures reported on the Schedule are reported on the accrual basis of accounting.  Such expenditures are recognized following the cost principles contained in the Uniform Guidance, wherein certain types of expenditures are not allowable or are limited as to reimbursement.",
                TYPE_ID=1,
                DBKEY="123456789",
                AUDITYEAR="2022",
}
```
and then in your `def _mock_notes_*`:
```
notes.append(self.MockNote(policy));
notes.append(self.MockNote(rate));
```
or whatever is needed.